### PR TITLE
Allow generate from useaga to work in TypOfIsNot expressions

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes internal bug 1130960

### CODE REVIEW

**Comment by jmarolf** [comment]
@srivatsn @tmeschter @shyamnamboodiripad @heejaechang @mavasani @JohnHamby Could you take a look please?

File: src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateVariable/GenerateVariableTests.vb [hunk 1]
```diff
@@ -909,5 +909,41 @@ NewLines("Module C \n Sub Test() \n If TypeOf [|B|] Is String Then \n End If \n
...
+NewLines("Imports System \n Imports System.Collections.Generic \n Imports System.Linq \n Module Program \n Private ReadOnly Prop As TypeOfIsNotDerived \n Sub M() \n If TypeOf Prop IsNot TypeOfIsNotDerived Then \n End If \n End Sub \n End Module"),
+index:=2)
+        End Sub
+
+        <WorkItem(1130960)>
+        <Fact(), Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)>
+        Public Sub TestGenerateLocalInTypeOfIsNot()
+            Test(
+NewLines("Imports System \n Imports System.Collections.Generic \n Imports System.Linq \n Module Program \n Sub M() \n If TypeOf [|Prop|] IsNot TypeOfIsNotDerived Then \n End If \n End Sub \n End Module"),
+NewLines("Imports System \n Imports System.Collections.Generic \n Imports System.Linq \n Module Program \n Sub M() \n Dim Prop As TypeOfIsNotDerived = Nothing \n If TypeOf Prop IsNot TypeOfIsNotDerived Then \n End If \n End Sub \n End Module"),
```
Comment (shyamnamboodiripad) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateVariable/GenerateVariableTests.vb]: What if ```TypeOfIsNotDerived``` is not defined. Do we offer the Generate Type fix in that case?

**Comment by shyamnamboodiripad** [comment]
:+1:

**Comment by srivatsn** [comment]
LGTM

File: src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateVariable/GenerateVariableTests.vb (hunk 1 above)
Comment (jmarolf) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateVariable/GenerateVariableTests.vb]: Yes, its hard to tell when its all on one line, but TypeOfIsNotDerived  is not defined in these tests.

**Pull request merged**